spyderlib/spyder.py

```diff
@@ -1242,6 +1250,10 @@ def post_visible_setup(self):
         self.extconsole.setMinimumHeight(0)
 
         if not self.light:
+            # Update lock status of dockidgets (panes)
+            self.lock_dockwidgets_action.setChecked(self.dockwidgets_locked)

```

Comment:
This line is the one that will make sure the action is selected according to the status when spyder was closed